ompi/mca/osc/ucx/osc_ucx_comm.c

```diff
@@ -797,20 +731,21 @@ static int
 do_atomic_compare_and_swap(const void *origin_addr, const void *compare_addr,
                            void *result_addr, struct ompi_datatype_t *dt,
                            int target, uint64_t remote_addr,
-                           ompi_osc_ucx_module_t *module)
+                           ompi_osc_ucx_module_t *module, ompi_win_t *win)
 {
     int ret;
     bool lock_acquired = false;
     size_t dt_bytes;
     opal_common_ucx_wpmem_t *mem = module->mem;
     if (!module->acc_single_intrinsic) {
-        ret = start_atomicity(module, target, &lock_acquired);
+        /* Start atomicity by acquiring acc lock  */
+        ret = ompi_osc_state_lock(module, target, &lock_acquired);
         if (ret != OMPI_SUCCESS) {
             return ret;
         }
     }
 
-    CHECK_DYNAMIC_WIN(remote_addr, module, target, ret);
+    CHECK_DYNAMIC_WIN(remote_addr, module, target, ret, win);
```

Comment:
I think there is still an issue: if `!module->acc_single_intrinsic` we take the state lock and then would take the state lock again in `get_dynamic_win_info`. Maybe check in `get_dynamic_win_info` if the state lock was taken (by passing `lock_acquired` from above)?

Reply:
Good catch. Yes, we need to check that. I'll add this.